### DISCUSSION TITLE
sktime with prophet for multivariate time series

### DISCUSSION THREAD

**Discussion by Nikhitha74** [alan-turing-institute/sktime#4142] [Q&A]
Hi, 
I have a doubt regarding using multivariate data for forecast and regression using prophet. I am using X  which has values ds,Open ,High,low and last price values and y =ds, Close, where ds is the date column. I have coded it as below: However using plot series I am getting error. I am not sure how to proceed. Kindly help me out.

[errorsktime.docx](https://github.com/sktime/sktime/files/10486885/errorsktime.docx)

**Comment by Nikhitha74** [comment]
Thank you for the solution given to an earlier similar question. I realised that the date column should be indexed. Now it is working fine. However when I try to set the forecasting horizon, it is throwing error.
I want to forecast for train data as well as test data.

y_train, y_test, X_train, X_test = temporal_train_test_split(y, X, test_size=25)
# Forecasting horizon, same as test data
fh1 = ForecastingHorizon(y_test.index, is_relative=False)

forecaster = Prophet(
    seasonality_mode="multiplicative",
    yearly_seasonality=False,
    weekly_seasonality=False,
    daily_seasonality=False,
    seasonality_prior_scale = 10,
    changepoint_prior_scale = 0.05,
   )

forecaster.fit(y=y_train,X=X_train)
y_pred = forecaster.predict(X=X_test,fh=fh1)#For test set

The code above is working fine.
However when i apply similar steps to train dataset, i  get error.

fh2=ForecastingHorizon(y_train.index, is_relative=False)
This gives to error
---> 1 fh2=ForecastingHorizon(y_train.index, is_relative=False)

File ~\anaconda3\envs\for_Sktime\lib\site-packages\sktime\forecasting\base\_fh.py:286, in ForecastingHorizon.__init__(self, values, is_relative, freq)
    277 def __init__(
    278     self,
    279     values: Union[VALID_FORECASTING_HORIZON_TYPES] = None,
   (...)
    284 
    285     # values to pd.Index self._values
--> 286     values = _check_values(values)
    287     self._values = values
    289     # infer freq from values, if available
    290     # if not, infer from freq argument, if available

File ~\anaconda3\envs\for_Sktime\lib\site-packages\sktime\forecasting\base\_fh.py:136, in _check_values(values)
    134 # check values does not contain duplicates
    135 if len(values) != values.nunique():
--> 136     raise ValueError(
    137         "Invalid `fh`. The `fh` values must not contain any duplicates."
    138     )
    140 # return sorted values
    141 return values.sort_values()

ValueError: Invalid `fh`. The `fh` values must not contain any duplicates.

When i checked for duplicates in the y_train index i could not find any duplicates.
train_pred=forecaster.predict(X=X_train,fh2)#For train set This line of code hence does not work.

Can someone help me in this?

**Comment by fkiraly** [comment]
@Nikhitha74, could you provide complete self-contained code, i.e., code that contains all the imports, and optimally uses some dummy data (instead of data that needs to be loaded)?

This would help us diagnose the issue much quicker.

**Reply by Nikhitha74** [reply]
Thank you for the quick response. I was using some dummy data and found out that the date index indeed had two duplicate values in the training set. Once i removed them, i could run it successfully.
Thank you once again.

One more question may I ask you. Can we use lstm for regression task with sktime temporal split. i need to compare the results so obtained with sktime prophet .

Kindly give solution.

**Reply by Nikhitha74** [reply]
Hi ,

I tried using lstm with make reduction .But I am confused about how to use the lstm model as the input need to consider window size and no of  features  we are using..
How do we go about this? I am working on a project where I want to compare lstm  results with prophet for stock prices. Any help would be deeply appreciated.

**Reply by fkiraly** [reply]
That depends on what you mean by "LSTM model".
Is it a temporal model, is it sth you wrote yourself, or is it coming from another python package?
